test/transforms/test_half_hop.py

```diff
@@ -0,0 +1,46 @@
+import torch
+
+from torch_geometric.data import Data
+from torch_geometric.transforms import HalfHop
+
+
+def test_half_hops():
+    edge_index = torch.tensor([[0, 1, 1, 2], [1, 0, 2, 1]])
```

Comment:
Can we add self loops here to make sure they remain intact.

Reply:
Done. Thank you for your review!